src/parser/md.rs

```diff
@@ -290,25 +256,16 @@ impl MdParser {
                                 if block.source.is_empty()
                                     && line.trim().starts_with(&self.comment_start)
                                 {
-                                    let trim = line.trim()[self.comment_start.len()..].trim();
-                                    let name = self.parse_name(trim, false);
-                                    match name {
-                                        Ok((name, vars, defaults)) => {
-                                            let hidden = name.starts_with(&self.hidden_prefix);
-                                            let name = if hidden {
-                                                &name[self.hidden_prefix.len()..]
-                                            } else {
-                                                &name[..]
-                                            };
-                                            block.name = Some(name.to_string());
-                                            block.hidden = hidden;
-                                            block.vars = vars;
-                                            block.defaults = defaults;
-                                        }
-                                        Err(error) => {
-                                            return Some(Parse::error(error, line_number));
-                                        }
+                                    let name = line.trim()[self.comment_start.len()..].trim();
+                                    let hidden = name.starts_with(&self.hidden_prefix);
```

Comment:
This would probably be a good place to use [`str::strip_prefix`](https://doc.rust-lang.org/stable/std/primitive.str.html#method.strip_prefix)

Reply:
Did it like this, but not sure if that's the solution you had in mind:

```rust
let mut name = line.trim()[self.comment_start.len()..].trim();
let hidden = if let Some(stripped) = name.strip_prefix(&self.hidden_prefix) {
    name = stripped;
    true
} else {
    false
};
```

Reply:
I think we can avoid the temporaries, e.g.
```rust
block.name = line.trim()[self.comment_start.len()..].trim();
if let Some(stripped) = name.strip_prefix(&self.hidden_prefix)
{
    block.name = stripped;
    block.hidden = true;
}
```
and there is another instance of the strip-prefix pattern above concerning `comment_start`.

Reply:
Ok, thanks. Will do that before the next merge.

Reply:
> and there is another instance of the strip-prefix pattern above concerning comment_start.

Too bad that `if` and `if let` can't be combined...

Reply:
This is under consideration but it looks a funky admittedly. You can achieve a similar effect using guard pattern even now though:
```rust
match foo {
  Some(foo) if foo.baz && qux => { ... }
  _ => (),
}
```

Reply:
Ok, but that makes it complicated again due to necessity for trimming. I'll leave the outer `if` as is, but changed to avoid the temporaries (although a bit more compicated, as block.name is an Option):

```rust
if let Some(stripped) = name.strip_prefix(&self.hidden_prefix) {
    block.name = Some(stripped.to_string());
    block.hidden = true;
} else {
    block.name = Some(name.to_string());
};
```